packages/next/next-server/server/next-server.ts

```diff
@@ -1007,6 +1014,34 @@ export default class Server {
     const isProduction = !this.renderOpts.dev
     const isDynamicPathname = isDynamicRoute(pathname)
     const didRespond = isResSent(res)
+
+    // we lazy load the staticPaths to prevent the user
+    // from waiting on them for the page to load in dev mode
+    let staticPaths = this.staticPathsCache[pathname]
```

Comment:
Let's eliminate this caching for now.

Reply:
I thought we decided making this blocking would be not ideal as `getStaticPaths` may have expensive fetching so we wanted to do it lazily to not block the user but still catch any errors in `getStaticPaths`?

Reply:
IIRC we floated this idea in #next-internal and other team members were not fond of the idea.